scripts/technical-debt-metrics.sh

```diff
@@ -15,6 +15,28 @@ IFS=$'\n\t'
 currCommit="${1:-"$(git rev-parse HEAD)"}"
 refCommit="${2:-"$(git log --pretty=%H --since="$(date -I -d 'last week')" | tail -n -1)"}"
 
+## `computeDiff input` assumes that input consists of lines of the form `value|description`
+## where `value` is a number and `description` is the statistic that `value` reports.
+## `value` is non-negative, if it refers to the current commit and it is negative otherwise.
+## The output is of the form `|current value|difference|description|`.
+computeDiff () {
+  awk -F'|' 'BEGIN{con=1}{
+      # order keeps track of maintaining the final order of the counters the same as the input one
+      # rdict stores the difference of the current value minus the old value
+      # curr stores the current value, making sure that the number is non-negative and does not start with `-`
+      if(rdict[$2] == "") {order[con]=$2; con++}
+      if((0 <= $1+0) && (!($1 ~ "-"))) {curr[$2]=$1}
+      rdict[$2]+=$1+0
+    } END {
+    # loop over the "sorted" index in `order`
+    for(ind=1; ind<=con-1; ind++) {
+      # retrieve the description of the counter
+      val=order[ind]
+      # print `|current value|difference|name of counter|`
+      printf("|%s|%s|%s|\n", curr[val], rdict[val], val)}
```

Comment:
Here we could add a check that, if `curr[val]` is `0` (that is, this statistics is likely no longer relevant), then we skip the `printf` step.

This would allow to have "dormant" statistics that would reactivate as soon as they become relevant again.

Reply:
I'd prefer not to do this. If there were some bug introduced while the counter was 0, and then it became nonzero, then we would possibly not know for a long time since we would assume it was dormant.

Reply:
Sure, I did not implement this because I had my reservations as well.  I'll mark as resolved!